### PR TITLE
fix(optimizer): add catalog and db to schema

### PR DESCRIPTION
**Pull request by barakalon**
Yikes.

@tobymao

### CODE REVIEW

File: sqlglot/optimizer/qualify_columns.py [hunk 1]
```diff
@@ -202,10 +202,21 @@ def _get_selectable_columns(name, selectables, schema):
 
     # If referencing a table, return the columns from the schema
     if isinstance(selectable, exp.Table):
+        catalog_name = selectable.text("catalog")
```
Comment (tobymao) [inline sqlglot/optimizer/qualify_columns.py:205]: can we make it so catalog and db are optional? so gracefully handle the cases where the schema object has different levels of nesting? i presume in many cases, we'd only pass in tables, and in most cases only db/table in schema. i don't think we'll ever use catalog

**Comment by barakalon** [comment]
@tobymao An elegant interface that supports a simple dictionary _and_ catalog/database nested tables was surprisingly tricky.

I don't like introspecting the provided dictionary too deeply - that can get quite nasty.

So this just assumes that a simple dictionary is only a mapping of tables.

If a user wants databases/catalogs, we'll need other implementations of the abstract `Schema` class. 

I need databases in Minerva, but I also want to lazy load table metadata, so implementing the Schema class is actually better.

File: tests/test_docs.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import inspect
 
 import sqlglot
+import sqlglot.optimizer
```
Comment (tobymao) [inline tests/test_docs.py:6]: why is this needed? we’re the test broken?

**Review by tobymao — APPROVED** [review]
i think this is reasonable, can you implement the catalog and db versions with nested dicts? should be trivial right?

File: tests/test_docs.py (hunk 2 above)
Comment (barakalon) [reply tests/test_docs.py:6]: Yeah, it wasn't running locally. Might be a setup quirk on my side... but this felt harmless.

**Comment by barakalon** [comment]
Alright, this now supports:
1. {table: {col: type}}
2. {db: {table: {col: type}}}
3. {catalog: {db: {table: {col: type}}}}
